tests/auth_tests/test_management.py

```diff
@@ -905,6 +906,63 @@ def test(self):
 
         test(self)
 
+    @mock.patch.dict(os.environ, {
+        'DJANGO_SUPERUSER_PASSWORD': 'test_password',
+        'DJANGO_SUPERUSER_USERNAME': 'test_superuser',
+        'DJANGO_SUPERUSER_EMAIL': 'joe@somewhere.org',
+        'DJANGO_SUPERUSER_FIRST_NAME': 'foo',
+    })
+    def test_environment_variable_non_interactive(self):
+        call_command('createsuperuser', interactive=False, stdout=StringIO())
+        user = User.objects.get(username='test_superuser')
+        self.assertEqual(user.email, 'joe@somewhere.org')
+        self.assertTrue(user.check_password('test_password'))
+        # Environment variables are ignored for non-required fields.
+        self.assertEqual(user.first_name, '')
+
+    @mock.patch.dict(os.environ, {
+        'DJANGO_SUPERUSER_USERNAME': 'superuser',
+        'DJANGO_SUPERUSER_EMAIL': 'joe1@somewhere.org',
+    })
+    def test_ignore_environment_variable_non_interactive(self):
+        """
+        Environment variables are ignored in non-interactive mode, if provided.
+        """
+        call_command(
+            'createsuperuser',
+            interactive=False,
+            username='test_superuser',
+            email='joe@somewhere.org',
+            stderr=StringIO(),
+        )
+        user = User.objects.get(username='test_superuser')
+        self.assertEqual(user.email, 'joe@somewhere.org')
+        self.assertFalse(user.has_usable_password())
+
+    @mock.patch.dict(os.environ, {
+        'DJANGO_SUPERUSER_PASSWORD': 'password2',
+        'DJANGO_SUPERUSER_USERNAME': 'superuser',
+        'DJANGO_SUPERUSER_EMAIL': 'joe1@somewhere.org',
+    })
+    def test_ignore_environment_variable_interactive(self):
+        """
+        Environment variables are ignored in interactive mode.
+        """
+        @mock_inputs({'password': 'test_password'})
+        def test(self):
+            call_command(
+                'createsuperuser',
+                interactive=True,
+                username='test_superuser',
+                email='joe@somewhere.org',
+                stderr=StringIO(),
```

Comment:
In this test `stdin=MockTTY()` is still required.

Reply:
Sorry for missing.